arviz_dashboard/forest.py

```diff
@@ -0,0 +1,217 @@
+            multi_select, thre_slider, rope_slider, 
+            data_variable, coor_variable
+        ):
+            # generate graph
+            data = []
+            for model_ in multi_select:
+                data.append(self.idatas_cmp[model_])
+            # add rope
+            rope = {}
+            school = {}
+            school["school"] = coor_variable
+            school["rope"] = rope_slider
+            rope[data_variable] = [school]
+            # print(rope)
+            forest_plt = az.plot_forest(
+                data,
+                model_names=multi_select,
+                rope=rope,
+                kind="forestplot",
+                hdi_prob=thre_slider,
+                backend="bokeh",
+                figsize=(9, 9),
+                show=False,
+                combined=True,
+                colors="cycle",
+            )
+            return forest_plt[0][0]
+
+        @pn.depends(
+            multi_select.param.value,
+            thre_slider.param.value,
+            truncate_checkbox.param.value,
+            ridge_quant.param.value,
+            op_slider.param.value,
+        )
+        def get_ridge_plot(
+            multi_select,
+            thre_slider,
+            truncate_checkbox,
+            ridge_quant,
+            op_slider,
+        ):
+            # calculate the ridgeplot_quantiles
+            temp_quant = list(ridge_quant)
+            quant_ls = temp_quant
+            quant_ls.sort()
+            avg_quant = sum(temp_quant) / 2
+            if quant_ls[0] < 0.5 and quant_ls[1] > 0.5:
+                quant_ls.append(0.5)
+                quant_ls.sort()
+            else:
+                quant_ls.append(avg_quant)
+                quant_ls.sort()
+
+            # generate graph
+            data = []
+            for model_ in multi_select:
+                data.append(self.idatas_cmp[model_])
+
+            ridge_plt = az.plot_forest(
+                data,
+                model_names=multi_select,
+                kind="ridgeplot",
+                hdi_prob=thre_slider,
+                ridgeplot_truncate=truncate_checkbox,
+                ridgeplot_quantiles=quant_ls,
+                ridgeplot_overlap=op_slider,
+                backend="bokeh",
+                figsize=(9, 9),
+                show=False,
+                combined=True,
+                colors="white",
+            )
+            return ridge_plt[0][0]
+
+        plot_result_1 = pn.Column(
+            pn.WidgetBox(
+                "add rope",
+                pn.Row(
+                    self.param.model,
+                    self.param.data_variable,
+                    self.param.coor_variable,
+                ),
+                rope_slider,
+            ),
+            get_forest_plot,
+        )
+        plot_result_2 = pn.Column(
+            pn.Row(truncate_checkbox),
+            pn.Row(ridge_quant, op_slider),
+            get_ridge_plot,
+        )
+        # show up
+        display(
```

Comment:
You are assuming this is run in a Juptyer environment. You should comment about that fact.